### PR TITLE
feat(backend): initialize ORM, crawling, GraphQL endpoints

### PR DESCRIPTION
**Pull request by barjin**
A barrel PR joining multiple things together. The +4000 LOC is caused mostly by the automatic lockfile update, real changes are around 700 LOC altogether. 

More details in GitHub comments with the code:

### CODE REVIEW

File: packages/backend/src/crawling/Crawler.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  });
+
+  static async crawl(execution: Execution) {
+    await this.workerPool.run(
+      {
+        executionId: execution.id,
+      },
+      { name: "run" },
+    );
+  }
```
Comment (barjin) [inline packages/backend/src/crawling/Crawler.ts:22]: Creates a "worker pool" (an abstraction over the Node.JS worker threads) with the given file as the "multithreaded" code.

Calling `workerPool.run` with parameters (`executionId`) runs the given file in a separate thread (worker thread).

File: packages/backend/src/crawling/worker.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import path from "path";
```
Comment (barjin) [inline packages/backend/src/crawling/worker.ts:1]: This file runs in the worker thread (the `workerWrapper.js` is a way of making `ts-node` load the Typescript file in workers.

File: packages/backend/src/data-source.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import "reflect-metadata";
```
Comment (barjin) [inline packages/backend/src/data-source.ts:1]: This is an ORM config file

File: packages/backend/src/routes/executions.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import { Router } from "express";
```
Comment (barjin) [inline packages/backend/src/routes/executions.ts:1]: This file exports a router (all the GET / PUT / POST handlers) for the `/executions` backend path.

**Review by viktorija-panovska — APPROVED** [review]
Looks good to me

**Pull request merged**